src/test/java/org/jabref/logic/integrity/IntegrityCheckTest.java

```diff
@@ -145,6 +149,94 @@ void testEntryIsUnchangedAfterChecks() {
         assertEquals(clonedEntry, entry);
     }
 
+    /**
+     * testCheckersForBibTex use injection to test the checkers in integrity check of BibTex
+     *
+     * @throws NoSuchFieldException   when using getDeclaredField in integrityCheck.getClass().getDeclaredField("entryCheckers")
+     * @throws IllegalAccessException when using Get method in entryCheckersField.get(integrityCheck)
+     */
+    @Test
+    void testCheckersForBibTex() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
These tests are a bit complicated. In addition, it's not useful to test a private fields or methods. https://dzone.com/articles/dont-test-private-methods
You can implicitly test if the function was called. Have a look at the IntegrityCheckTest class
It tests the integrity checker to produce correct results, e.g. it tests that the check is working and produces the correct result.

Reply:
Yeah, and if I directly use the return result to check the IntegrityCheck, It maybe still complicated (since every check I would manage to give an error entry to get the return result set.)! 
So maybe it is better to write the test in the submodel of integrity check to test the checkers? (It means that I will simply remove these 2 tests).
Hope for your reply!

Reply:
Yes, just delete the two tetst classes. The UTF8Checkers test is fine.

Reply:
OK,  I have removed the unwieldy Junit tests